### PR TITLE
Module for replacing empty nodes by containing tags

### PR DESCRIPTION
**Pull request by Minstel**

### CODE REVIEW

File: spec/replace-empty-node-spec.js [hunk 1]
```diff
@@ -0,0 +1,289 @@
+var tidy = require('../');
+
+describe('Replacing empty dom nodes with containing tags', function() {
```
Comment (jasny) [inline spec/replace-empty-node-spec.js:3]: Please also add tests for common cases

```html
<p>{{#tag}}</p>
<p>Hello world</p>
<p>{{/tag}}</p>
```

should become `{{#tag}}<p>Hello world</p>{{/tag}}`

and 

```html
<p>{{#tag}}</p>
<p>Hello world</p>
<p>{{/tag}} {{^tag}}</p>
<p>Good night</p>
<p>{{/tag}}</p>
```
should become `{{#tag}}<p>Hello world</p>{{/tag}} {{^tag}}<p>Good night</p>{{/tag}}`

File: scripts/utils.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+
+// Set dependencies
+var jsdom = require('jsdom');
+jsdom.defaultDocumentFeatures = {
+    FetchExternalResources: false,
+    ProcessExternalResources: false
+};
+
+var Node = require('../node_modules/jsdom/lib/jsdom/living/generated/Node.js');
```
Comment (jasny) [inline scripts/utils.js]: Can't you use `require('jsdom/lib/jsdom/living/generated/Node.js')`?

**Review by jasny — CHANGES_REQUESTED** [review]

**Comment by Minstel** [comment]
```
<p>{{#tag}}</p>
<p>Hello world</p>
<p>{{/tag}} {{^tag}}</p>
<p>Good night</p>
<p>{{/tag}}</p>
```

To correctly handle `<p>{{/tag}} {{^tag}}</p>`, second step (moving tags) is required, when closing tag is first moved out of node. Currently replacing nodes hanles the following cases: 

- `<p>{{#tag}}</p>`
- `<p>{{/tag}}</p>`
- `<p>{{^tag}} {{/tag}}</p>`

This is related to current algo - while handling current tags section, it does not look for sibling tags (for better performance), and just consideres them as some data.
I did corresponding change to readme.

**Comment by jasny** [comment]
@Minstel You should treat `{{^tag}}` exactly the same as `{{#tag}}`. A `^` just means 'not'.

If a `<p>` contains only start and/or end tags, than remove `<p>`. It doesn't order or number of start tags doesn't matter. Doing step 2 without removing this, means we'll end up with empty paragraphs.

**Pull request merged**